typed-store/src/rocks/tests.rs

```diff
@@ -99,6 +99,80 @@ fn test_multi_get() {
     assert_eq!(result[2], None);
 }
 
+#[test]
+fn test_non_conflicting_insert() {
+    let db = DBMap::open(temp_dir(), None, None).expect("Failed to open storage");
+
+    // non-conflicting works on an empty key
+    db.non_conflicting_insert(&123456789, &"123456789".to_string())
+        .expect("Failed to non-conflicting insert");
+    assert_eq!(
+        Some("123456789".to_string()),
+        db.get(&123456789).expect("Failed to get")
+    );
+
+    // non-conflicting does not work on a set key
+    db.insert(&23456, &"23456".to_string())
+        .expect("Failed to insert");
+    db.non_conflicting_insert(&23456, &"65432".to_string())
```

Comment:
Maybe this is just me but I wish there was a way to return a value indicating whether the write succeeded or not.  To return OK when a write didn't happen seems to be, non ideal from API perspective.